docs/llm-provider/llm_provider_huggingface.md

```diff
@@ -0,0 +1,168 @@
+# Hugging Face Provider
+
+Hugging Face provides access to thousands of open-source models that can run locally or via their inference API.
+
+## Supported Models
+
+Popular models for code analysis:
+
+| Model                       | Size | Type    | Best For                |
+|-----------------------------|------|---------|-------------------------|
+| `microsoft/DialoGPT-medium` | 345M | General | Conversational analysis |
+| `Salesforce/codet5-base`    | 220M | Code    | Code understanding      |
+| `microsoft/codebert-base`   | 125M | Code    | Code representation     |
+| `bigcode/starcoder`         | 15B  | Code    | Advanced code tasks     |
+
+## Setup
+
+### Prerequisites
+
+Hugging Face models require additional dependencies:
+
+```bash
+# Install as PRE_COMMAND in .mega-linter.yml
+pip install langchain-huggingface transformers torch
+```
+
+Or use the optional dependency group:
+```bash
+pip install megalinter[huggingface]
+```
```

Comment:
You mentioned in another file that this project uses uv, should we show the usage with it? It's not incompatible, pip here is still universal and the venvs created by uv can be changed by pip afterwards without problem. However, we are just inconsistent

Reply:
torch is 200 MB heavy so i didn't want to install it in all MegaLinter images, but install it "on the fly" using a PRE_COMMAND for HuggingFace users

The pre-commansd doc uses pip, do you confirm it will work ?

https://megalinter.io/beta/config-precommands/